aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/helper/HeaderUtils.java

```diff
@@ -0,0 +1,321 @@
+package com.amazonaws.services.lambda.runtime.events.helper;
+
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.*;
+
+/**
+ * This class exposes some utility methods to work with request values such as headers
+ * and query string parameters.
+ */
+public class HeaderUtils {
+
+    static final String HEADER_KEY_VALUE_SEPARATOR = "=";
+    static final String HEADER_VALUE_SEPARATOR = ";";
+    static final String HEADER_QUALIFIER_SEPARATOR = ",";
+    static final String ENCODING_VALUE_KEY = "charset";
+
+
+    //-------------------------------------------------------------
+    // Methods
+    //-------------------------------------------------------------
+
+    /**
+     * Given a map of key/values query string parameters from API Gateway, creates a query string as it would have
+     * been in the original url.
+     * @param parameters A Map&lt;String, String&gt; of query string parameters
+     * @param encode Whether the key and values should be URL encoded
+     * @param encodeCharset Charset to use for encoding the query string
+     * @return The generated query string for the URI
+     */
+    public static String generateQueryString(Map<String, List<String>> parameters, boolean encode, String encodeCharset)
+            throws Exception {
+
+        if (parameters == null || parameters.size() == 0) {
+            return null;
+        }
+
+        StringBuilder queryStringBuilder = new StringBuilder();
+
+        try {
+            for (String key : parameters.keySet()) {
+                for (String val : parameters.get(key)) {
+                    queryStringBuilder.append("&");
+                    if (encode) {
+                        queryStringBuilder.append(URLEncoder.encode(key, encodeCharset));
+                    } else {
+                        queryStringBuilder.append(key);
+                    }
+                    queryStringBuilder.append("=");
+                    if (val != null) {
+                        if (encode) {
+                            queryStringBuilder.append(URLEncoder.encode(val, encodeCharset));
+                        } else {
+                            queryStringBuilder.append(val);
+                        }
+                    }
+                }
+            }
+        } catch (UnsupportedEncodingException e) {
+            throw new Exception("Invalid charset passed for query string encoding", e);
+        }
+        String queryString = null;
+
+        queryString = queryStringBuilder.toString();
+        queryString = queryString.substring(1); // remove the first & - faster to do it here than adding logic in the Lambda
+        return queryString;
+    }
+
+    /**
+     * Parses a header value using the default value separator "," and qualifier separator ";".
+     * @param headerValue The value to be parsed
+     * @return A list of SimpleMapEntry objects with all of the possible values for the header.
```

Comment:
what is SimpleMapEntry? I see the method is returning List of HeaderValue

Reply:
Old comment that I forgot to update. Fixed.